IconThemePlugin.js

```diff
@@ -0,0 +1,44 @@
+const path = require('path');
+
+class IconThemePlugin {
+  constructor(options = {}) {
+    this.metadataPath = options.metadataPath;
+  }
+
+  apply(compiler) {
+    const pluginName = IconThemePlugin.name;
+    const {webpack} = compiler;
+    const {RawSource} = webpack.sources;
+
+    compiler.hooks.emit.tap(pluginName, (compilation) => {
+      const icons = compilation.getAssets().map(
+        (asset) => asset.name.split('/').pop()
+      );
+
+      const metadataFile = `${this.metadataPath || compiler.context}/metadata.json`;
+      const metadata = require(metadataFile);
+
+      const iconEntries = icons.map((file) => [
+        file.substr(0, file.lastIndexOf('.')),
+        file.substr(file.lastIndexOf('.') + 1, file.length)
+      ]);
+
+      const sortedIcons = Object.fromEntries(
```

Comment:
Technically, you don't need to do this, because `.sort()` actually mutates the array (yuck... I know).

So

```
iconEntries.sort((a, b) => a[0].localeCompare(b[0]))
```

And then 

```
      metadata.icons = Object.fromEntries(
        Object.entries(iconEntries).filter(
          ([key, value]) => imageTypes.includes(value)
        )
      );
```

Gets you the same result, just a little bit less code :)

Reply:
Also, this might be a bit nicer:

```
iconEntries.sort(([a], [b]) => a.localeCompare(b))

```